### PR TITLE
Move more code out of //flutter/sky/engine

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: sky/engine/bindings/flutter_dart_state.cc [hunk 1]
```diff
@@ -34,7 +33,7 @@ FlutterDartState::~FlutterDartState() {
 }
 
 FlutterDartState* FlutterDartState::CreateForChildIsolate() {
-  return new FlutterDartState(isolate_client_, url_);
+  return new FlutterDartState(isolate_client_);
```
Comment (chinmaygarde) [inline sky/engine/bindings/flutter_dart_state.cc]: Maybe have this return a unique_ptr?

**Comment by chinmaygarde** [comment]
lgtm

File: sky/engine/bindings/flutter_dart_state.cc (hunk 1 above)
Comment (abarth) [inline sky/engine/bindings/flutter_dart_state.cc]: The pointer ends up being stored in side Dart_Isolate and manually deleted in the isolate shutdown callback.  I could make this return a unique_ptr, but we'd have to release it right away to put it in the isolate.

**Pull request merged**